libnmstate/appliers/linux_bridge.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import six
+
+
+def get_slaves_from_state(state, default=()):
+    ports = state.get('bridge', {}).get('port')
+    if ports is None:
+        return default
+    return [p['name'] for p in ports]
+
+
+def set_bridge_ports_metadata(master_state, slave_state):
+    _set_common_slaves_metadata(master_state, slave_state)
+
+    ports = master_state.get('bridge', {}).get('port', [])
+    port = next(
+        six.moves.filter(lambda n: n['name'] == slave_state['name'], ports),
+        {}
+    )
+    slave_state['_brport_options'] = port
+
+
+def _set_common_slaves_metadata(master_state, slave_state):
```

Comment:
This seems to be a generic method that is not specific to linux bridges, can it maybe be moved to a different place?

Reply:
Yes, I think you are right. I left/duplicated it here because I have no better place to place it in.
It should get out of here when we split the metadata handlers out of the netapplier.